### ISSUE TITLE
Freeze on hero picking

### ISSUE THREAD

**Issue by wedonn** [ValveSoftware/Dota-2#2384]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/): https://gist.github.com/wedonn/cd5e6414e756d9ecac0f721bff6b010f
* Have you checked for system updates?: Yes
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: Yes

#### Please describe your issue in as much detail as possible:

In approximately 1/10 matches on the hero selection menu game visually freezes, but I still can hear all sounds and (if very lucky) can pick a hero. After the hero picking stage ends, I just have a black screen with no sound. When I manually restart the game and press "connect", it still leaves me with a black screen and I'm still disconnected. This already gave me about 6 low priorities.

![image](https://github.com/ValveSoftware/Dota-2/assets/85764364/1b156779-564f-4d9e-a52a-807d8b63a873)
(not my screenshot, but it looks like this)

**Comment by wedonn** [comment]
Please help. Game is literally unplayable because I basically have endless low priority

**Comment by wedonn** [comment]
The simillar bug is happening again, but now the game actually starts after (blindly) picking a hero. I guess it's like an overlay of the first frame before all players connect. After approximately 50 seconds the bug dissapears, but at that time, I have to rather pick a random hero I see first or just get banned.

**Comment by Cubits11** [comment]
Yeah, I have been trying to play this game for a week now. It's been so long since I played this game, and just wanted to hop and enjoy the nostalgia. Tried everything, but nothing worked for me. Is this a Steam issue, or our computers are not powerful which doesn't even make sense.

**Comment by bnbean** [comment]
Yeah I quit the game because of this actually for months.. maybe hoping it would have been fixed but yeah.. I am not able to play the game.. Because it freezes at hero selection 

My system specs are Win 11
ClevoP170EM 
i7 3820QM
GTX 980M 
16GB DDR3

**Comment by vininim** [comment]
This is kind happening here too, but the game resumes after some time has elapsed (usually can't ban and miss first pick phase/gold).

Here is a video of it:
https://github.com/ValveSoftware/Dota-2/assets/19626/6aaae348-d183-4591-b5c1-14cc5e4b57b3

Here is Steam Hardware Info: 
https://gist.github.com/vininim/fdff46035b37ca883fdd5e114f4b678a
